motoman_driver/MotoPlus/IoServer.c

```diff
@@ -97,6 +97,7 @@ void Ros_IoServer_StartNewConnection(Controller* controller, int sd)
 			controller->sdIoConnections[connectionIndex] = INVALID_SOCKET;
 			controller->tidIoConnections[connectionIndex] = INVALID_TASK;
 			Ros_Controller_SetIOState(IO_FEEDBACK_FAILURE, TRUE);
+			mpSetAlarm(8004, "MOTOROS FAILED TO CREATE TASK", 7);
```

Comment:
Yes, I've given each mpSetAlarm a different subcode (pretty much in the order they are created), so that if the alarm 8004 occurs, it could help identify which task wasn't started.  Generally speaking, I doubt that we would dig down to that level because whenever that 8004 occurs, it's probably because of parameters settings or you have multiple versions of the MotoRos running.  So I didn't go as far as creating constants for them.